metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -1125,13 +1125,11 @@ private Errors validateAlterPartitionData(
             return INVALID_REQUEST;
         }
 
-        List<Integer> ineligibleReplicas = partitionData.newIsr().stream()
-            .filter(replica -> !isEligibleReplica.apply(replica))
-            .collect(Collectors.toList());
+        List<IneligibleReplica> ineligibleReplicas = ineligibleReplicasForIsr(newIsr);
         if (!ineligibleReplicas.isEmpty()) {
             log.info("Rejecting AlterPartition request from node {} for {}-{} because " +
                     "it specified ineligible replicas {} in the new ISR {}.",
-                    brokerId, topic.name, partitionId, ineligibleReplicas, partitionData.newIsr());
```

Comment:
nit: We may want to revert this change in order to remain consistent with the other log line in this method.